### ISSUE TITLE
Logger dependency conflict with Vagrant 2.4.2

### ISSUE THREAD

**Issue by carlosaya** [vagrant-libvirt/vagrant-libvirt#1831]
**Describe the bug**
Vagrant 2.4.2 was released 3 days ago, which apparently changes the underlying version of the logger and results in a dependency conflict when installing vagrant-libvirt.

```
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Vagrant failed to properly resolve required dependencies. These errors can commonly be caused by misconfigured plugin installations

         Conflicting dependency chains:
           logger (= 1.6.1), 1.6.1 activated

         Conflicting dependency chains:
           logger (= 1.6.1), 1.6.1 activated

         Conflicting dependency chains:
           logger (= 1.6.1), 1.6.1 activated
           logger (= 1.6.1), 1.6.1 activated
```

Reverting back to Vagrant 2.4.1 works fine.

**Comment by koberbe** [comment]
+1 with the same Vagrant version

**Comment by sir-ragna** [comment]
Seems like a fix was merged into Vagrant for this https://github.com/hashicorp/vagrant/issues/13527

**Comment by carlosaya** [comment]
> Seems like a fix was merged into Vagrant for this [hashicorp/vagrant#13527](https://github.com/hashicorp/vagrant/issues/13527)

Thanks. I did check there first but didn't look at closed issues 🤦‍♂️

**Comment by rgl** [comment]
[Vagrant 2.4.3](https://github.com/hashicorp/vagrant/releases/tag/v2.4.3) fixed the installation for me. But vagrant now shows the following warning:

```console
$ vagrant destroy -f
[fog][WARNING] Unrecognized arguments: libvirt_ip_command
==> default: Pruning invalid NFS exports. Administrator privileges will be required...
==> default: Removing domain...
==> default: Deleting the machine folder
```

**Comment by LKHN** [comment]
@rgl,
I'm experiencing the same issue. This is due to version `0.13.0` of https://github.com/fog/fog-libvirt. Which was released yesterday. I think this deserves a separate issue.

Cc: @electrofelix

**Comment by ajpenner** [comment]
The issues happened for me this morning using Vagrant 2.4.3.

**Comment by cvoltz** [comment]
Install works for me with:

- vagrant 2.4.3
- vagrant-libvirt 0.12.2
- ruby 3.3.3

I do see the warning about `libvirt_ip_command` though when using vagrant.

**Comment by ajpenner** [comment]
> Install works for me with:
> 
> - vagrant 2.4.3
> - vagrant-libvirt 0.12.2
> - ruby 3.3.3
> 
> I do see the warning about `libvirt_ip_command` though when using vagrant.

Yes, I got there through a chain. I agree it is just a warning that comes through, not an error.

**Comment by trinitronx** [comment]
> [Vagrant 2.4.3](https://github.com/hashicorp/vagrant/releases/tag/v2.4.3) fixed the installation for me. But vagrant now shows the following warning:
> 
> ```
> $ vagrant destroy -f
> [fog][WARNING] Unrecognized arguments: libvirt_ip_command
> ==> default: Pruning invalid NFS exports. Administrator privileges will be required...
> ==> default: Removing domain...
> ==> default: Deleting the machine folder
> ```

I'm seeing this too with `vagrant` `2.4.3`, `vagrant-libvirt` `0.12.2`, and `fog-libvirt` `0.13.1`.  The host system is running `libvirt` `10.10.0`.

> @rgl, I'm experiencing the same issue. This is due to version `0.13.0` of https://github.com/fog/fog-libvirt. Which was released yesterday. I think this deserves a separate issue.
> 
> Cc: @electrofelix

Yes, it looks like this was introduced in fog/fog-libvirt@97450dd, which dropped support for `libvirt < 1.2.8`.  It looks like the `libvirt_ip_command` was used as a fallback in case the DHCP method was not supported.  Seems that it's [a warning][1] caused by the `libvirt_ip_command` method being [removed from the `recognized` options][2].

So long as the newer DHCP method for leasing and then finding the libvirt "domain"'s (_VM_'s) IP is working, this warning can probably be safely ignored. 

[1]: https://github.com/fog/fog-core/blob/fbcb048c4ab6bf790bdb6158b6fc89a06c6f9b0d/lib/fog/core/service.rb#L250
[2]: https://github.com/fog/fog-libvirt/commit/97450dd#diff-aea1b857f83e8b2aa12130a08e1d1f3ce51b5665b0e44024dd2213c9ee555634L9

**Comment by alexventuraio** [comment]
I got the same error as the original poster in MacOS `Sequoia 15.3`.

- I installed vagrant following the [official documentation](https://developer.hashicorp.com/vagrant/install#darwin) but I think that is wrong because is actually installing `v2.4.2` so someone should update it.
- Also not sure if the [Brew](https://formulae.brew.sh/cask/vagrant) command shown there is the official one or maybe an old old one that needs to be updated as well:
  - Instead of using this:
    ```ruby
    brew tap hashicorp/tap
    brew install hashicorp/tap/hashicorp-vagrant
    ```
  - The good one for `v2.4.3` is:
    ```ruby
    brew install vagrant
    ```
After using the lates command, I got Vagrant successfully installed and I was able to run `vagrant plugin install vagrant-vmware-desktop` successfully as well.
